signetdev/host/signetdev.c

```diff
@@ -52,9 +53,15 @@ int signetdev_device_num_root_blocks()
 	return 1;
 }
 
-int signetdev_priv_cmd_payload_size()
+unsigned int signetdev_priv_cmd_payload_size()
 {
-	return signetdev_device_block_size() + CMD_PACKET_HEADER_SIZE + 1;
+    int bs = signetdev_device_block_size();
+    if (bs == -1) {
+        perror("Invalid Signet device!");
```

Comment:
I would only want signet_base to print something if it hit a catastrophic error and had to close the program. Otherwise I would want the client to handle any printing.

Reply:
Ok, I agree, no prints here.

Normally we would use -1 to propagate errors. Do you really want to switch to `unsigned int` and return 0?